docs/api/wanchain.md

```diff
@@ -1,5 +1,255 @@
 # API / Wanchain
 
+## Deploy gateway
+
+Deploy gateway allows for delegating deployment of an `AssetLedger` smart contract to a third party and transfering value in the same atomic transaction.
+
+To perform a delegate deploy through `DeployGateway`, you need to follow its flow:
+
+1. Maker (address creating the delegate deploy) defines the deploy (defines the data for the `AssetLedger` that will be created, who will be the owner of that `AssetLedger` and what amount of tokens will be sent to what address in exchange).
+2. Maker generates the deploy claim and signs it (`claim` functions).
+3. Maker approves value transfer.
+4. Maker sends deploy and signature to the Taker trough arbitrary channel.
+6. Taker performs the deploy.
+
+`Deploy` class is responsible for defining what will happen in the atomic swap. We support two different deploy configurations which we will call a fixed deploy and a dynamic deploy.
+
+In fixed deploy, the Taker of the deploy (its wallet address) is known, and we want to make an atomic deploy specifically with him and only him. For this, we need to set `deploy.takerId`.
+
+In dynamic deploy, we do not care who performs the deploy. In this case, we do not set `deploy.takerId`. Now any account (wallet) will be able to perform such deploy and will automatically become its Taker.
```

Comment:
In dynamic deploy, the deploy can be performed by anyone. For this reason, we do not set `deploy.takerId`. This will enable any account (wallet) to perform such deploy and automatically become its Taker.